tests/copy-paste.js

```diff
@@ -0,0 +1,780 @@
+            util.loadXML("");
+            paste([
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default"],
+                ["", "Text", "text", "text"],
+            ]);
+            util.selectAll();
+            eq(mod.copy(), [
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default"],
+                ["/question1", "Text", "text", "text"],
+            ]);
+        });
+
+        it("should paste question with / id", function () {
+            util.loadXML("");
+            paste([
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default"],
+                ["/", "Text", "text", "text"],
+            ]);
+            util.selectAll();
+            eq(mod.copy(), [
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default"],
+                ["/question2", "Text", "text", "text"],
+            ]);
+        });
+
+        it("should not paste item into or after text", function () {
+            util.loadXML("");
+            paste([
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default"],
+                ["/text", "Text", "text", "text"],
+            ]);
+            util.selectAll();
+            paste([
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default"],
+                ["/select/item", "Item", "item1", "item1"],
+            ], ["Cannot insert Choice into or after Text"]);
+            util.selectAll();
+            eq(mod.copy(), [
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default"],
+                ["/text", "Text", "text", "text"],
+            ]);
+        });
+
+        it("should not overwrite itext with auto-id", function () {
+            util.loadXML("");
+            paste([
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default"],
+                ["/text", "Text", "auto", "auto"],
+            ]);
+            util.clickQuestion("text");
+            paste([
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default", "labelItext"],
+                ["/text2", "Text", "non", "non", "text-label"],
+            ]);
+            util.selectAll();
+            eq(mod.copy(), [
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default", "labelItext"],
+                ["/text", "Text", "auto", "auto", "null"],
+                ["/text2", "Text", "non", "non", "text-label"],
+            ]);
+            call("createXML");
+            var text = util.getMug("text"),
+                text2 = util.getMug("text2");
+            // NOTE which one gets auto-renamed is somewhat arbitrary
+            assert.equal(text.p.labelItext.autoId, true);
+            assert.equal(text.p.labelItext.id, "text-label");
+            assert.equal(text2.p.labelItext.autoId, false);
+            assert.equal(text2.p.labelItext.id, "text-label2");
```

Comment:
Should this test also make sure that the actual text for text2 is still 'non' and not 'auto'?